### ISSUE TITLE
Wayland HiDPI: winit decoration overhangs the contents by 2x in examples

### ISSUE THREAD

**Issue by myfreeweb** [gfx-rs/wgpu-rs#51]
Something is not right about the hidpi factor handling:

![cube-fs8](https://user-images.githubusercontent.com/208340/62121915-3e03ff80-b2cd-11e9-90bf-3a483332666c.png)

**Comment by kaimast** [comment]
Are you using gnome shell? It might be the same `winit` problem I am having.

https://github.com/rust-windowing/winit/issues/1509

**Comment by myfreeweb** [comment]
No, this is Wayfire. Other winit apps like Alacritty work great.

**Comment by kaimast** [comment]
Weird..I just tested and don't have this issue with alacritty either.
